src/auth/ProtectedRoute.tsx

```diff
@@ -3,7 +3,7 @@ import { RootStateOrAny, useSelector } from "react-redux";
 import { Navigate } from "react-router-dom";
 
 function ProtectedRoute({ children }: { children: JSX.Element }) {
-    const auth = useSelector((state: RootStateOrAny) => state.isAuthenticated);
+    const auth = useSelector((state: RootStateOrAny) => state.auth.isAuthenticated);
```

Comment:
you can create separate file to store small functions which can be used in useSelector 

export const isAuth = (state: RootStateOrAny) => state.auth.isAuthenticated;

and use them everywhere in code to avoid duplicates

 const auth = useSelector(isAuth);